File: algorithms/selectionsort.go
package main

import "fmt"

type artistData struct {
	Name  string
	Plays int
}

func selectionSort(artists *[]artistData) {
	array_length := len(*artists)
	for i := 0; i < (array_length - 1); i++ {
		smaller_index := i
		for j := i + 1; j < array_length; j++ {
			if (*artists)[j].Plays < (*artists)[smaller_index].Plays {
				smaller_index = j
			}
		}
		if smaller_index >= 0 {
			var smaller_play_count artistData = (*artists)[smaller_index]
			(*artists)[smaller_index] = (*artists)[i]
			(*artists)[i] = smaller_play_count
		}
	}
}

func main() {
	artists := []artistData{
		{"Artist A", 400},
		{"Artist B", 50},
		{"Artist C", 150},
		{"Artist D", 300},
	}
	selectionSort(&artists)
	fmt.Println(artists)
}

File: algorithms/selectionsort.py
def selection_sort(artists):
    length_of_arr = len(artists)
    for i in range(length_of_arr):
        bigger_index = None
        play_count = artists[i]["plays"]
        for j in range(i, length_of_arr):
            if artists[j]["plays"] > play_count:
                bigger_index = j
        if bigger_index:
            bigger_play_count = artists[j]
            artists[bigger_index] = artists[i]
            artists[i] = bigger_play_count


if __name__ == "__main__":
    artists = [
        {"name": "Artist A", "plays": 400},
        {"name": "Artist B", "plays": 50},
        {"name": "Artist C", "plays": 150},
        {"name": "Artist D", "plays": 300},
    ]
    selection_sort(artists)
    print(artists)